File: javalin/src/main/java/com/github/zjor/ujapi/Application.java
package com.github.zjor.ujapi;

import com.github.zjor.ujapi.config.JavalinModule;
import com.github.zjor.ujapi.config.MongoModule;
import com.github.zjor.ujapi.ext.guice.LoggingModule;
import com.google.inject.Guice;
import com.google.inject.Injector;
import io.javalin.Javalin;

public class Application {
    public static void main(String[] args) {
        Injector injector = Guice.createInjector(
                new LoggingModule(),
                new MongoModule(),
                new JavalinModule()
        );

        injector.getInstance(Javalin.class).start(8080);
    }
}

File: src/main/java/com/github/zjor/ujapi/Application.kt
package com.github.zjor.ujapi

import com.github.zjor.ujapi.config.JacksonModule
import com.github.zjor.ujapi.config.JavalinModule
import com.github.zjor.ujapi.config.MongoModule
import com.github.zjor.ujapi.ext.guice.LoggingModule
import com.google.inject.Guice
import io.javalin.Javalin

object Application {
    @JvmStatic
    fun main(args: Array<String>) {
        val injector = Guice.createInjector(
                LoggingModule(),
                MongoModule(),
                JacksonModule(),
                JavalinModule()
        )
        val port = System.getenv("PORT")?.toInt() ?: 8080
        injector.getInstance(Javalin::class.java).start(port)
    }
}